biscuit-auth/src/bwk.rs

```diff
@@ -0,0 +1,209 @@
+use std::convert::TryFrom;
+
+use chrono::{DateTime, FixedOffset};
+use serde::{Deserialize, Serialize};
+
+use crate::{error, PublicKey};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(into = "BiscuitWebKeyRepr")]
+#[serde(try_from = "BiscuitWebKeyRepr")]
+pub struct BiscuitWebKey {
+    pub public_key: PublicKey,
+    pub key_id: u32,
+    pub issuer: Option<String>,
+    pub expires_at: Option<DateTime<FixedOffset>>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct BiscuitWebKeyRepr {
+    pub algorithm: String,
+    pub key_bytes: String,
+    pub key_id: u32,
+    pub issuer: Option<String>,
+    pub expires_at: Option<DateTime<FixedOffset>>,
+}
+
+impl From<BiscuitWebKey> for BiscuitWebKeyRepr {
+    fn from(value: BiscuitWebKey) -> Self {
+        BiscuitWebKeyRepr {
+            algorithm: "ed25519".to_string(),
+            key_bytes: value.public_key.to_bytes_hex(),
+            key_id: value.key_id,
+            issuer: value.issuer,
+            expires_at: value.expires_at,
+        }
+    }
+}
+
+impl TryFrom<BiscuitWebKeyRepr> for BiscuitWebKey {
+    type Error = error::Format;
+
+    fn try_from(value: BiscuitWebKeyRepr) -> Result<Self, Self::Error> {
+        if value.algorithm != "ed25519" {
+            return Err(error::Format::DeserializationError(format!(
+                "deserialization error: unexpected key algorithm {}",
+                value.algorithm
+            )));
+        }
+
+        let public_key = PublicKey::from_bytes_hex(&value.key_bytes)?;
+
+        Ok(BiscuitWebKey {
+            public_key,
+            key_id: value.key_id,
+            issuer: value.issuer,
+            expires_at: value.expires_at,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Why are the tests in the same file? Is it really convenient mixing test/prod code?